File: 卡码/代码随想录/C++/二叉树/04-二叉搜索树属性/LeetCode_0098.cpp
//98. ÑéÖ¤¶ş²æËÑË÷Ê÷
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
	TreeNode* pre=nullptr;
public:
  bool isValidBST(TreeNode* root) {
		if(!root) return true;
		bool left=isValidBST(root->left);
		if(pre!=nullptr&&pre->val>=root->val) return false;
		pre=root;
		bool right=isValidBST(root->right);
		return left&&right;
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_04_二叉搜索树属性/LeetCode_0098.java
//98. 验证二叉搜索树
package com.kurumi.binarytree._04_二叉搜索树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0098 {

}
class Solution {
  TreeNode pre=null;
  public boolean isValidBST(TreeNode root) {
    if(root==null) return true;
    boolean left=isValidBST(root.left);
    if(pre!=null&&pre.val>=root.val) return false;
    pre=root;
    boolean right=isValidBST(root.right);
    return left&&right;
  }
}
